src/main/java/net/minecraftforge/fmp/block/TileCoverable.java

```diff
@@ -0,0 +1,169 @@
+package net.minecraftforge.fmp.block;
+
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.network.NetworkManager;
+import net.minecraft.network.play.server.SPacketUpdateTileEntity;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.AxisAlignedBB;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.fmp.capabilities.ISlottedCapabilityProvider;
+import net.minecraftforge.fmp.capabilities.MultipartCapabilityHelper;
+import net.minecraftforge.fmp.microblock.IMicroblock;
+import net.minecraftforge.fmp.microblock.IMicroblockContainerTile;
+import net.minecraftforge.fmp.microblock.MicroblockContainer;
+import net.minecraftforge.fmp.multipart.IMultipart;
+import net.minecraftforge.fmp.multipart.PartSlot;
+
+/**
+ * An implementation of {@link TileMultipartContainer} and {@link IMicroblockContainerTile} that acts as a microblock
+ * container.<br/>
+ * Extend this class if you want a custom TileEntity for your {@link BlockCoverable}.
+ */
+public class TileCoverable extends TileEntity implements IMicroblockContainerTile, ISlottedCapabilityProvider
+{
+    
+    private MicroblockContainer container;
+
+    public TileCoverable()
+    {
+    }
+
+    @Override
+    public World getWorldIn()
+    {
+        return getWorld();
+    }
+
+    @Override
+    public BlockPos getPosIn()
+    {
+        return getPos();
+    }
+
+    @Override
+    public MicroblockContainer getMicroblockContainer()
+    {
+        return container != null ? container : (container = new MicroblockContainer(this));
+    }
+
+    @Override
+    public boolean canAddMicroblock(IMicroblock microblock)
+    {
+        return true;
+    }
+
+    @Override
+    public void onMicroblocksChanged()
+    {
+    }
+
+    @Override
+    public boolean hasCapability(Capability<?> capability, EnumFacing facing)
+    {
+        if (super.hasCapability(capability, facing)) { return true; }
```

Comment:
Typically this is the other way around. The specific handler, and then the super, because the super contains the handlers added by the events from other mods.